jsonnet/kube-state-metrics.jsonnet

```diff
@@ -33,6 +33,25 @@ local tlsVolumeName = 'kube-state-metrics-tls';
                 caFile: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
                 serverName: 'server-name-replaced-at-runtime',
               },
+              // Drop the "instance" and "pod" labels since we're runinng only
+              // one instance of kube-state-metrics. The "instance" label can't
+              // be dropped at the discovery stage because otherwise Prometheus
+              // will default its value to the address being scraped.
+              // This is to avoid excessive series churn when
```

Comment:
Yes it may happen when the kube-state-metrics pod's IP address changes (e.g. on restart/redeploy) but it should be rare. The normal path should be:
1. Prometheus detects that the old pod is gone, stops the scrape loop and waits for (2.1 x scrape interval) before inserting the stale markers.
2.  Prometheus creates a new scrape loop for the new kube-state-metrics pod's address.
3. At the next scrape interval, Prometheus scrapes the new target before the stale markers get inserted.
4. Prometheus tries to insert stale markers for the old loop but fails with an out-of-order error because series have already been inserted at stage 3.

If the new scrape loop doesn't complete before step 4 then it will fail because of out-of-order samples.